devtools/gradle/gradle-application-plugin/src/main/java/io/quarkus/gradle/QuarkusPlugin.java

```diff
@@ -244,6 +250,24 @@ public void execute(Task test) {
                         testNative.setShouldRunAfter(Collections.singletonList(tasks.findByName(JavaPlugin.TEST_TASK_NAME)));
                     });
 
+                    SourceSet intTestSourceSet = sourceSets.create(INTEGRATION_TEST_SOURCE_SET_NAME);
+                    intTestSourceSet.setCompileClasspath(
+                            intTestSourceSet.getCompileClasspath()
+                                    .plus(mainSourceSet.getOutput())
+                                    .plus(testSourceSet.getOutput()));
+                    intTestSourceSet.setRuntimeClasspath(
+                            intTestSourceSet.getRuntimeClasspath()
+                                    .plus(mainSourceSet.getOutput())
+                                    .plus(testSourceSet.getOutput()));
+
+                    tasks.register(INTEGRATION_TEST_TASK_NAME, Test.class, intTestTask -> {
+                        intTestTask.setGroup("verification");
+                        intTestTask.setDescription("Runs Quarkus integration tests");
+                        intTestTask.dependsOn(quarkusBuild);
+                        intTestTask.setClasspath(intTestSourceSet.getRuntimeClasspath());
+                        intTestTask.setTestClassesDirs(intTestSourceSet.getOutput().getClassesDirs());
+                    });
```

Comment:
Do you need `intTestTask.setShouldRunAfter(List.of(tasks.findByName(JavaPlugin.TEST_TASK_NAME)));`?